### PR TITLE
Checkout

### PR DESCRIPTION
**Pull request by scmountain**

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class OrdersController < ApplicationController
+
+  def new
+    @item_names = @cart.cart_items
+    @item_prices = @cart.cart_item_prices
+  end
+
+  def create
+    order = Order.new(order_date: Time.now)
```
Comment (blackknight75) [inline app/controllers/orders_controller.rb]: This was our finest moment last night! We can do some refactoring later but I think this setup for create is going to give us a lot of additional query capabilities for our admin dashboard.

File: app/models/cart.rb [hunk 2]
```diff
@@ -39,4 +39,16 @@ def self.cart_contents(contents)
         Item.find(item_id)
     end
   end
+
+  def cart_items
```
Comment (blackknight75) [inline app/models/cart.rb]: This worked out so great! Not having this logic in the view is perfect!

File: db/migrate/20170227013245_create_order_items.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateOrderItems < ActiveRecord::Migration
+  def change
+    create_table :order_items do |t|
+      t.references :item, index: true, foreign_key: true
+      t.references :order, index: true, foreign_key: true
+      t.integer :quantity
+      t.string :status
```
Comment (blackknight75) [inline db/migrate/20170227013245_create_order_items.rb]: Good call on the status being on the joins table. Being able to have a status for each line item makes this way more dynamic in terms of order management.

File: config/routes.rb [hunk 4]
```diff
@@ -20,7 +20,7 @@
 
   resources :items, only: [:new, :show]
   resources :users, only: [:show, :create]
-  # resources :admins
-
+  resources :orders, only: [:new, :create, :show]
```
Comment (scmountain) [inline config/routes.rb]: @carmer @s-espinosa, would there be benefit to nesting order under user? technically the session[:cart] belongs to a user... not sure how to ask this.

Comment (s-espinosa) [inline config/routes.rb]: I think one of the main benefits of nesting is having access to the `has_many` part of the relationship in params coming from the routes. That's not really the case with user (assume you're using `current_user` in these cases). Let's sleep on it and discuss tomorrow.

**Pull request merged**